src/PageObject/AdminPage.php

```diff
@@ -0,0 +1,148 @@
+<?php
+namespace PaulGibbs\WordpressBehatExtension\PageObject;
+
+use SensioLabs\Behat\PageObjectExtension\PageObject\Page;
+
+/**
+ * Page object representing a generic admin page.
+ *
+ * This class houses methods for navigating around the WordPress admin and
+ * acts as a base for most page-specific admin page objects.
+ */
+class AdminPage extends Page
+{
+    /**
+     * Returns the text in the header tag.
+     *
+     * The first h1 element is used, or first h2 element if it is not present.
+     *
+     * @return string The text in the header tag.
+     */
+    public function getHeaderText()
+    {
+        $header = $this->getHeaderElement();
+        $header_text = $header->getText();
+        $header_link = $header->find('css', 'a');
+
+        // The page headers can often incude an 'add new link'. Strip that out of the header text.
+        if ($header_link) {
+            $header_text  = trim(str_replace($header_link->getText(), '', $header_text));
+        }
+
+        return $header_text;
+    }
+
+    /**
+     * Asserts the text in the header tag matches the given text.
+     *
+     * @throws \Exception
+     */
+    public function assertHasHeader($expected)
+    {
+        $actual = $this->getHeaderText();
+        if ($expected !== $actual) {
+            throw new \Exception(sprintf('Expected page header "%s", found "%s".', $expected, $actual));
+        }
+    }
+
+    /**
+     * Get the text in the header tag/
+     *
+     * The first h1 element is used, or first h2 element if it is not present.
+     *
+     * @throws \Exception
+     */
+    private function getHeaderElement()
+    {
+        // h2s were used prior to 4.3/4 and h1s after
+        // @see https://make.wordpress.org/core/2015/10/28/headings-hierarchy-changes-in-the-admin-screens/
+        $header2 = $this->find('css', '.wrap > h2');
+        $header1 = $this->find('css', '.wrap > h1');
+
+        if ($header1) {
+            return $header1;
+        } elseif ($header2) {
+            return $header2;
+        }
+
+        throw new \Exception('Header could not be found');
+    }
+
+    /**
+     * Click a link within the page header tag.
+     *
+     * @param string $link Link text.
+     */
+    public function clickLinkInHeader($link)
+    {
+        $header = $this->getHeaderElement();
+        $header->clickLink($link);
+    }
+
+    /**
+     * Returns the AdminMenu element.
+     *
+     * @return AdminMenu
+     */
+    public function getMenu()
+    {
+        return $this->getElement('Admin menu');
+    }
+
+    /**
+     * Modified Page::isOpen() function which throws an exception on failure
+     * @see https://github.com/sensiolabs/BehatPageObjectExtension/issues/57
+     * @param array $url_parameters
+     * @return boolean
+     * @throws SensioLabs\Behat\PageObjectExtension\PageObject\Exception\UnexpectedPageException
```

Comment:
I think if I wanted to put a note for a Human, i'd put it in the function's long description. Putting stuff in these at-tags makes it easier for PHPDoc to extract that, and do its work. As far as I'm concerned, that kind of document parser/IDE ought to be smart enough to analyse the code to see if an exception is thrown, or have enough context that it knows a called function can throw an exception -- and with that innate understanding, it can go build what output it wants.

Just my 5 cents